subprojects/core/src/integTest/groovy/org/gradle/api/tasks/CopyPermissionsIntegrationTest.groovy

```diff
@@ -193,5 +194,30 @@ class CopyPermissionsIntegrationTest extends AbstractIntegrationSpec {
         testTargetFile.canWrite()
     }
 
+    @Requires(TestPrecondition.FILE_PERMISSIONS)
+    @Issue('https://github.com/gradle/gradle/issues/2639')
+    def "excluded files' permissions should be ignored"() {
```

Comment:
:x: This test does not fail if I run it without your changes (OS X, both Java 8 or Java 7). Maybe it does not reproduce the issue correctly?

Reply:
My fault, I confused some file permission stuff. Now I fixed it. So sorry for that. Please check it again when you're free.